urllib3/transport_security.py

```diff
@@ -28,11 +28,26 @@ def __init__(self, transport_security_store=None):
 
     def validate_connection(self, conn):
         """
-        Enforce connection security checks such as HSTS and HPKP.
+        Enforce connection security checks such as HSTS. This can be called
+        multiple times on the same connection before each request,
+        since new HSTS headers may be processed in the meantime.
+
         This is a stub, to be implemented later.
 
         :param conn:
             A :class:`urllib3.connection.HTTPConnection` instance.
+
+        """
+
+    def validate_ssl_connection(self, conn):
```

Comment:
I don't love this name along with `validate_connection`: they're both fundamentally about validating TLS. I appreciate that naming here is really hard, but I'm still not sure about this.

Would it be better to have `validate_new_connection` and `validate_established_connection`? Or possibly just `validate_hpkp` and `validate_hsts`? At least those names are clear and descriptive: even if they're overly specific, we can always rename them later when we have a better feel for how these will fit together.